lib/src/builder.dart

```diff
@@ -94,8 +92,14 @@ class GeneratorBuilder extends Builder {
     var generatedOutputs =
         await _generate(library, generators, buildStep).toList();
 
-    var contentBuffer = new StringBuffer();
+    // Don't output useless files.
+    //
+    // NOTE: It is important to do this check _before_ checking for valid
```

Comment:
Would prefer if this was protected by a test rather than a comment. Can you file an issue so we can followup later?